packages/core/src/metrics/aggregator.ts

```diff
@@ -0,0 +1,165 @@
+import type {
+  Client,
+  ClientOptions,
+  MeasurementUnit,
+  MetricsAggregator as MetricsAggregatorBase,
+  Primitive,
+} from '@sentry/types';
+import { timestampInSeconds } from '@sentry/utils';
+import { DEFAULT_FLUSH_INTERVAL, MAX_WEIGHT, NAME_AND_TAG_KEY_NORMALIZATION_REGEX } from './constants';
+import { METRIC_MAP } from './instance';
+import type { MetricBucket, MetricType } from './types';
+import { getBucketKey, sanitizeTags } from './utils';
+
+/**
+ * A metrics aggregator that aggregates metrics in memory and flushes them periodically.
+ */
+export class MetricsAggregator implements MetricsAggregatorBase {
+  // TODO(@anonrig): Use FinalizationRegistry to have a proper way of flushing the buckets
+  // when the aggregator is garbage collected.
+  // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry
+  private _buckets: MetricBucket;
+
+  // Different metrics have different weights. We use this to limit the number of metrics
+  // that we store in memory.
+  private _bucketsTotalWeight;
+
+  // TODO(@anonrig): Use `setTimeout` instead of `setInterval` to be more accurate
```

Comment:
Why is this more accurate? What does the change here involve?

Reply:
Because it doesn't take into the account of the time required to send the request to Sentry. If the goal is to send metrics to Sentry every 10 seconds, and if that request takes 1 second, you buffer 1 second worth of data into the second iteration.

Reply:
Ahh I see. Adding to transport queue is considered fire and forget though to not block on return value of response, so I think finding correct `setTimeout` is nontrivial.